src/compiler/checker.ts

```diff
@@ -3743,6 +3743,10 @@ namespace ts {
             return result;
         }
 
+        function createOriginType(flags: TypeFlags): Type {

```

Comment:
Is there a reason to not just use `createType`? Is it to avoid setting a `type.id` or adding it to the `typeCatalog`? This will make the numbers we report in diagnostics from `Checker.getTypeCount` inaccurate (since we don't increment `typeCount`).

Reply:
Origin types aren't real types because they violate normalization constraints we generally assume to be true. Also, they're just an alternate (denormalized) form of another type. For these reasons I'm not including them in the type count or type catalog.

Reply:
Its still something we probably want to report in diagnostics in case we want to help customers investigate performance issues in the future. If we want the counts separate, we could add an `originTypeCount` and expose the result either by having `Checker.getTypeCount()` return `typeCount + originTypeCount`, or return it via its own method on `Checker` and give it a separate diagnostic output.

Reply:
I agree, we should track these even if they live in a "shadow type realm" because we're still going to be doing work on them.

Reply:
I disagree. Origin types aren't real types, they're just digests we use for display purposes. It happens that storing them as in the same format as types is more convenient, but that's all. Reporting them in the type count is equivalent to saying every union or intersection type with for which to choose to track an origin is counted twice. I don't think that makes sense and I don't think there is any value in tracking them.